File: ntgcalls/src/models/call_network_state.cpp
//
// Created by Laky64 on 03/10/24.
//

#include <ntgcalls/models/call_network_state.hpp>

namespace ntgcalls {
    NetworkInfo::NetworkInfo(const ConnectionState connectionState, const Kind kind) : state(connectionState), kind(kind) {}
} // ntgcalls

File: android/app/src/main/java/org/pytgcalls/ntgcalls/CallNetworkState.java
package org.pytgcalls.ntgcalls;

public class CallNetworkState {
    public final Kind kind;
    public final State state;

    public enum Kind {
        NORMAL,
        PRESENTATION,
    }

    public enum State {
        CONNECTING,
        CONNECTED,
        FAILED,
        TIMEOUT,
        CLOSED
    }

    public CallNetworkState(Kind kind, State state) {
        this.kind = kind;
        this.state = state;
    }
}
